packages/design-system-react/src/components/icon/Icon.tsx

```diff
@@ -25,15 +34,10 @@ export const Icon: React.FC<IconProps> = ({
   );
 
   return (
-    <svg
+    <IconComponent
       className={mergedClassName}
       {...(props as React.SVGProps<SVGSVGElement>)}
       style={style}
-      xmlns="http://www.w3.org/2000/svg"
-      viewBox="0 0 512 512"
-      fill="currentColor"
-    >
-      <path d="m337 51l-171 0c-75 0-119 45-119 119l0 172c0 74 44 119 119 119l171 0c75 0 119-45 119-119l0-172c1-74-44-119-119-119z m-3 220l-67 0 0 67c0 8-7 15-15 15-9 0-16-7-16-15l0-67-66 0c-9 0-16-7-16-15 0-8 7-15 16-15l66 0 0-67c0-8 7-15 16-15 8 0 15 7 15 15l0 67 67 0c8 0 15 7 15 15 0 8-7 15-15 15z" />
-    </svg>
+    />
```

Comment:
Using dynamic component lookup (Icons[name]) instead of static SVG imports:
- Allows lazy loading of icons - only the icons that are actually used are included in the bundle
- Provides type safety through the IconName enum
- Makes it easy to add new icons without modifying the Icon component code
- Centralizes icon rendering logic and props handling in one place

typescript> const IconComponent = Icons[name];> if (!IconComponent) {> console.warn(Icon "${name}" not found);> return null;> }>
This pattern ensures consistent rendering and proper error handling while keeping our bundle size optimized.